### PR TITLE
office/pdftk: Download man page from upstream instead

### PR DESCRIPTION
**Pull request by isaackwy**
Previously, I downloaded the man page from gitlab (https://gitlab.com/pdftk-java/pdftk/-/raw/master/pdftk.1) and then manually added it to the SlackBuild folder. This was because the upstream man page could change (meaning that the md5sum of the man page would change as well).

I changed the SlackBuild so that it instead directly downloads the man page from https://gitlab.com/pdftk-java/pdftk/-/raw/v3.3.3/pdftk.1 (The SlackBuilds folder no longer stores a pdftk.1 file.) The 'raw' content should remain the same, as it downloads from 3.3.3, rather than from 'master'.

This will be my last pull request for this week.
More updates/pull requests from me next week.

EDIT: I just added a new commit. I included your (aclemons') suggestions in this pull request.

### CODE REVIEW

File: office/pdftk/pdftk.SlackBuild [hunk 1]
```diff
@@ -51,49 +43,38 @@ TMP=${TMP:-/tmp/SBo}
...
-
 set -e
 
 rm -rf $PKG
 mkdir -p $TMP $PKG $OUTPUT
 cd $TMP
-rm -rf $PRGNAM-$VERSION-dist
-unzip $CWD/$PRGNAM-$VERSION-src.zip
-cd $PRGNAM-$VERSION-dist
+
```
Comment (aclemons) [inline office/pdftk/pdftk.SlackBuild:51]: This chowns `$TMP` (by default `/tmp/SBo`). I don't think this is what you intended?

File: office/pdftk/pdftk.SlackBuild [hunk 2]
```diff
@@ -51,49 +43,38 @@ TMP=${TMP:-/tmp/SBo}
...
 find -L . \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
   -o -perm 511 \) -exec chmod 755 {} \; -o \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
-patch -p1 < $CWD/pdftk-2.02-use-gcc5.patch
+# Copy the binary
+mkdir -p $PKG/usr/share/java/$PRGNAM
+cp $CWD/$PRGNAM-all.jar $PKG/usr/share/java/$PRGNAM
```
Comment (aclemons) [inline office/pdftk/pdftk.SlackBuild:61]: All the steps in this slackbuild doing `cp $CWD/...` should use `install` instead imho so you can set the mode and owner/group explicitly.

Comment (isaackwy) [reply office/pdftk/pdftk.SlackBuild:61]: You mean the following?
install -Dm644 $CWD/$PRGNAM-all.jar $PKG/usr/share/java/$PRGNAM
install -Dm755 $CWD/$PRGNAM.sh $PKG/usr/bin/$PRGNAM
install -Dm644 $CWD/$PRGNAM.1 $PKG/usr/man/man1

File: office/pdftk/pdftk.SlackBuild (hunk 1 above)
Comment (isaackwy) [reply office/pdftk/pdftk.SlackBuild:51]: Then should I delete the following lines:
```
cd $TMP

chown -R root:root .
find -L . \
 \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
  -o -perm 511 \) -exec chmod 755 {} \; -o \
 \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
  -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
```

This does not cd to $TMP and does not chown $TMP.
I'll be back within 1 hour. Please confirm if you would like me to make the changes in this comment and in the one below.

File: office/pdftk/pdftk.SlackBuild (hunk 2 above)
Comment (aclemons) [reply office/pdftk/pdftk.SlackBuild]: Yes, exactly

**Comment by willysr** [comment]
@sbo-bot: build office/pdftk

**Comment by willysr** [comment]
this patch failed to apply cleanly
probably forgot to rebase?

**Comment by isaackwy** [comment]
How am I supposed to rebase, given that the first commit in this branch (updating pdftk to 3.3.3) was already merged but the second and third commits (additional SlackBuild edits) are not?

**Pull request closed without merging**